### DISCUSSION TITLE
msg.definition and DBIRTH

### DISCUSSION THREAD

**Discussion by hexfet75** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#18] [General]
In your guide ...
"_but in some cases its beneficial to configure the metrics via code. This can be done by configuring the metrics using the msg.definition attribute.DO NOT include definition is each message, as it will trigger rebirth each time a valid msg.definition is processed by the node."

If i dont include definition in each message i have got an error "The metric 'xxx/xxname' is not recognized by the device", while if i use UI definition its ok.

_

**Comment by thomassorensen2650** [comment]
Can you attach a small flow flow the demonstrate the issue? It must be a edge case not covered by the unit testing.

**Comment by hexfet75** [comment]
Sorry but I had a error in msg.definition, it's ok.

Thanks

Il sab 11 mar 2023, 19:58 thomassorensen2650 ***@***.***> ha
scritto:
